packages/server/app/Controllers/Http/AuthenticationController.test.ts

```diff
@@ -1095,3 +1099,354 @@ test.group('Validate password reset token', (group) => {
+            status: 'INVALID_TOKEN',
+        });
+    });
+
+    test('Returns an error if password is already used by the user', async (assert) => {
+        const request = createRequest();
+
+        const userPlainPassword = internet.password();
+        const user = await User.create({
+            uuid: datatype.uuid(),
+            nickname: internet.userName(),
+            email: internet.email(),
+            password: userPlainPassword,
+        });
+
+        const plainToken = generateToken();
+        const tokenType = await TokenType.findByOrFail(
+            'name',
+            TokenTypeName.enum.PASSWORD_RESET,
+        );
+        await user.related('tokens').create({
+            uuid: datatype.uuid(),
+            tokenTypeUuid: tokenType.uuid,
+            value: plainToken,
+            expiresAt: DateTime.local().plus({
+                minutes: 15,
+            }),
+        });
+
+        const requestBody: ResetPasswordRequestBody = {
+            token: plainToken,
+            email: user.email,
+            password: userPlainPassword,
+            authenticationMode: 'web',
+        };
+        const response = await request
+            .post('/authentication/reset-password')
+            .send(requestBody)
+            .expect(400);
+        const parsedResponseBody = ResetPasswordResponseBody.parse(
+            response.body,
+        );
+
+        assert.deepStrictEqual(parsedResponseBody, {
+            status: 'PASSWORD_ALREADY_USED',
+        });
+    });
+
+    test('Authenticates user with web mode if provided information are correct', async (assert) => {
+        const request = createRequest();
+
+        const user = await User.create({
+            uuid: datatype.uuid(),
+            nickname: internet.userName(),
+            email: internet.email(),
+            password: internet.password(),
+        });
+
+        const plainToken = generateToken();
+        const tokenType = await TokenType.findByOrFail(
+            'name',
+            TokenTypeName.enum.PASSWORD_RESET,
+        );
+        await user.related('tokens').create({
+            uuid: datatype.uuid(),
+            tokenTypeUuid: tokenType.uuid,
+            value: plainToken,
+            expiresAt: DateTime.local().plus({
+                minutes: 15,
+            }),
+        });
+
+        const requestBody: ResetPasswordRequestBody = {
+            token: plainToken,
+            email: user.email,
+            password: internet.password(),
+            authenticationMode: 'web',
+        };
+        const response = await request
+            .post('/authentication/reset-password')
+            .send(requestBody)
+            .expect(200);
+        const parsedResponseBody =
+            ResetPasswordSuccessfulWebAuthenticationResponseBody.parse(
+                response.body,
+            );
+
+        assert.deepStrictEqual(parsedResponseBody, {
+            status: 'SUCCESS',
+        });
+
```

Comment:
Suggestion: Unless I'm mistaken, you do not verify in your tests that the user password has changed ? We could compare at least that the hashed stored password has changed after the reset ?

Reply:
You're right! I'm going to update the successful path test.